src/main/java/com/google/netpcapanalysis/dao/BucketDaoImpl.java

```diff
@@ -0,0 +1,228 @@
+package com.google.netpcapanalysis.dao;
+
+import com.google.netpcapanalysis.models.Flagged;
+import com.google.netpcapanalysis.models.PCAPdata;
+import java.util.*; 
+import java.io.*;
+import java.lang.*;
+import java.util.Map.Entry;
+
+import io.pkts.PacketHandler;
+import io.pkts.Pcap;
+import io.pkts.buffer.Buffer;
+import io.pkts.packet.Packet;
+import io.pkts.packet.TCPPacket;
+import io.pkts.packet.UDPPacket;
+import io.pkts.packet.IPPacket;
+import io.pkts.protocol.Protocol;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+
+import com.google.netpcapanalysis.interfaces.dao.PCAPDao;
+import com.google.netpcapanalysis.interfaces.dao.PCAPParserDao;
+import com.google.netpcapanalysis.interfaces.dao.BucketDao;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+public class BucketDaoImpl implements BucketDao {
+  private ArrayList<PCAPdata> allPCAP; 
+  private ArrayList<PCAPdata> sortedPCAP; 
+  private String myip = "";
+
+  // Map to store <String Class, HashMap<> of protocols and frequencies >
+  private LinkedHashMap<String, HashMap<String, Integer>> bucketData;
+  
+  private String classA = "Class A";
+  private String classB = "Class B";
+  private String classC = "Class C";
+  private String classDE = "Class D & E";
+
+  public BucketDaoImpl(ArrayList<PCAPdata> packets) {
+    allPCAP = packets; 
+    orderIPs();
+    loadBuckets();    
+  }
+
+  public LinkedHashMap<String, HashMap<String, Integer>> getBuckets() {
+    return bucketData;
+  }
+ 
+  /* Put packets into MYIP, OUTIP format and order by destination IP (ascending). */
+  private void orderIPs(){
+    // find local IP 
+    findMyIP();
+
+    // sort all packets such that MYIP, OUTIP is the formatting
+    sortedPCAP = new ArrayList<PCAPdata>(); 
+    for (PCAPdata packet : allPCAP) {
+      String srcip = packet.source;
+      String dstip = packet.destination;
+      String outip = "";
+
+      if (srcip.equals(myip)) {
+        outip = dstip;
+      }
+      else {
+        outip = srcip;
+      }
+      
+      PCAPdata tempPCAP = new PCAPdata(myip, outip, "", "", packet.protocol, packet.size, packet.flagged, packet.frequency); 
+      sortedPCAP.add(tempPCAP);
+    }
+    // sort all packets in order of OUTIP
+    sortIPs();
+  }
+
+  public ArrayList<PCAPdata> getSortedPCAP(){
+    return sortedPCAP;
+  }
+
+  /* Find local ip address based on highest recurring IP address */  
+  private void findMyIP() {
```

Comment:
Isn't this function already in FrequencyDaoImpl.java? This is a useful function at will probably be used multiple times. I  added a modified version to my Traffic indicator pull request that looks like this `public String findMyIP(ArrayList<PCAPdata> allData)`, it would be nice to separated this function to allow for multiple uses